### PR TITLE
Moving to named parameters in format strings 

### PR DESCRIPTION
**Pull request by domenukk**
This was done using ./scripts/autofix.sh and supersedes #825

### CODE REVIEW

File: libafl/src/corpus/ondisk.rs [hunk 1]
```diff
@@ -176,7 +176,7 @@ where
                     break self.dir_path.join(file);
                 }
 
-                file = format!("{}-{}", &file_orig, ctr);
+                file = format!("{}-{ctr}", &file_orig);
```
Comment (domenukk) [inline libafl/src/corpus/ondisk.rs:179]: same here

File: libafl/src/executors/inprocess.rs [hunk 2]
```diff
@@ -1609,10 +1609,10 @@ where
                     // we can't do this from the parent, timerid is unique to each process.
                     libc::timer_create(libc::CLOCK_MONOTONIC, null_mut(), addr_of_mut!(timerid));
 
-                    println!("Set timer! {:#?} {:#?}", self.itimerspec, timerid);
+                    println!("Set timer! {:#?} {timerid:#?}", self.itimerspec);
                     let v =
                         libc::timer_settime(timerid, 0, addr_of_mut!(self.itimerspec), null_mut());
-                    println!("{:#?} {}", v, nix::errno::errno());
+                    println!("{v:#?} {}", nix::errno::errno());
```
Comment (domenukk) [inline libafl/src/executors/inprocess.rs:1615]: The lint can't do more than 1 I guess?

File: libafl/src/executors/inprocess.rs [hunk 3]
```diff
@@ -1609,10 +1609,10 @@ where
                     // we can't do this from the parent, timerid is unique to each process.
                     libc::timer_create(libc::CLOCK_MONOTONIC, null_mut(), addr_of_mut!(timerid));
 
-                    println!("Set timer! {:#?} {:#?}", self.itimerspec, timerid);
+                    println!("Set timer! {:#?} {timerid:#?}", self.itimerspec);
```
Comment (domenukk) [inline libafl/src/executors/inprocess.rs:1612]: ...

File: libafl/src/monitors/mod.rs [hunk 4]
```diff
@@ -39,14 +39,14 @@ pub enum UserStats {
...
+            UserStats::Number(n) => write!(f, "{n}"),
+            UserStats::Float(n) => write!(f, "{n}"),
+            UserStats::String(s) => write!(f, "{s}"),
             UserStats::Ratio(a, b) => {
                 if *b == 0 {
-                    write!(f, "{}/{}", a, b)
+                    write!(f, "{a}/{b}")
                 } else {
-                    write!(f, "{}/{} ({}%)", a, b, a * 100 / b)
+                    write!(f, "{a}/{b} ({}%)", a * 100 / b)
```
Comment (domenukk) [inline libafl/src/monitors/mod.rs:49]: or just complex types

File: libafl_cc/build.rs [hunk 5]
```diff
@@ -33,9 +33,9 @@ fn find_llvm_config_brew() -> Result<PathBuf, String> {
             if brew_cellar_location.is_empty() {
                 return Err("Empty return from brew --cellar".to_string());
             }
-            let cellar_glob = format!("{}/llvm/*/bin/llvm-config", brew_cellar_location);
+            let cellar_glob = format!("{brew_cellar_location}/llvm/*/bin/llvm-config");
             let glob_results = glob(&cellar_glob).unwrap_or_else(|err| {
-                panic!("Could not read glob path {} ({})", &cellar_glob, err);
+                panic!("Could not read glob path {} ({err})", &cellar_glob);
```
Comment (domenukk) [inline libafl_cc/build.rs:38]: ..

File: libafl_cc/build.rs [hunk 6]
```diff
@@ -122,20 +122,20 @@ fn build_pass(
...
+    println!("cargo:rerun-if-changed=src/{src_file}");
     if cfg!(unix) {
         assert!(Command::new(bindir_path.join("clang++"))
             .arg("-v")
             .args(cxxflags)
             .arg(src_dir.join(src_file))
             .args(ldflags)
             .arg("-o")
-            .arg(out_dir.join(format!("{}.{}", src_stub, dll_extension())))
+            .arg(out_dir.join(format!("{src_stub}.{}", dll_extension())))
```
Comment (domenukk) [inline libafl_cc/build.rs:133]: ..

File: libafl_cc/build.rs [hunk 7]
```diff
@@ -145,11 +145,11 @@ fn build_pass(
             .arg(format!(
                 "/OUT:{}",
                 out_dir
-                    .join(format!("{}.{}", src_stub, dll_extension()))
+                    .join(format!("{src_stub}.{}", dll_extension()))
```
Comment (domenukk) [inline libafl_cc/build.rs:148]: ...

File: libafl_concolic/symcc_runtime/build.rs [hunk 8]
```diff
@@ -208,8 +208,8 @@ fn write_symcc_rename_header(rename_header_path: &Path, cpp_bindings: &bindgen::
         .for_each(|val| {
             writeln!(
                 rename_header_file,
-                "#define {} {}{}",
-                &val, SYMCC_RUNTIME_FUNCTION_NAME_PREFIX, &val
+                "#define {} {SYMCC_RUNTIME_FUNCTION_NAME_PREFIX}{}",
```
Comment (domenukk) [inline libafl_concolic/symcc_runtime/build.rs:211]: ...

File: libafl_qemu/src/edges.rs [hunk 9]
```diff
@@ -229,7 +229,7 @@ where
             return None;
         }
     }
-    Some((hash_me(src as u64) ^ hash_me(dest as u64)) & (unsafe { EDGES_MAP_PTR_SIZE } as u64 - 1))
+    Some((hash_me(src) ^ hash_me(dest)) & (unsafe { EDGES_MAP_PTR_SIZE } as u64 - 1))
```
Comment (domenukk) [inline libafl_qemu/src/edges.rs:232]: @andreafioraldi could this cause issues on 32 bit systems?

File: libafl/src/corpus/ondisk.rs (hunk 1 above)
Comment (domenukk) [reply libafl/src/corpus/ondisk.rs:179]: Ah, borrows don't work here (?)

File: libafl/src/monitors/mod.rs (hunk 4 above)
Comment (domenukk) [reply libafl/src/monitors/mod.rs:49]: Arithmetics is not allowed in format strings either, apparently

File: libafl_cc/build.rs (hunk 6 above)
Comment (domenukk) [reply libafl_cc/build.rs:133]: Functions calls not allowed... :/

File: libafl_qemu/src/edges.rs (hunk 9 above)
Comment (domenukk) [reply libafl_qemu/src/edges.rs:232]: I think it will. Fixing

Comment (domenukk) [reply libafl_qemu/src/edges.rs:232]: Probably having `GuestAddr` be a `struct` would be more rust-y than a `type`

File: libafl_qemu/src/asan.rs [hunk 10]
```diff
@@ -108,7 +108,7 @@ extern "C" fn asan_giovese_printaddr(_addr: u64) -> *const u8 {
 #[no_mangle]
 unsafe extern "C" fn asan_giovese_populate_context(ctx: *mut CallContext, _pc: u64) {
     let ctx = ctx.as_mut().unwrap();
-    ctx.tid = libc::gettid() as i32;
+    ctx.tid = libc::gettid();
```
Comment (domenukk) [inline libafl_qemu/src/asan.rs:111]: Could this be an issue, cross-arch?

Comment (domenukk) [reply libafl_qemu/src/asan.rs:111]: https://docs.rs/libc/0.2.135/src/libc/unix/mod.rs.html#25

Seems good

**Review by tokatoka — APPROVED** [review]

File: libafl/src/corpus/ondisk.rs (hunk 1 above)
Comment (tokatoka) [reply libafl/src/corpus/ondisk.rs:179]: sounds like we'll fix this again when the lint got updated for borrow in the future 😅

Comment (domenukk) [reply libafl/src/corpus/ondisk.rs:179]: No, borrows don't work, and maybe never will. I understood this now, it's just a named parameter that gets filled with a variable of the same name automatically

**Pull request merged**